src/hooks/api/useApiCalls.ts

```diff
@@ -45,7 +60,91 @@ const useApiCalls = (): ApiCallsContextType => {
     [getValuesByChain]
   );
 
-  return { createType, stateCall };
+  const localTransfer = useCallback(
+    async (dispatchers, transfersData) => {
+      const { dispatchTransaction, dispatchMessage } = dispatchers;
+      const { receiverAddress, transferAmount, account } = transfersData;
+      const type = TransactionTypes.LOCAL_TRANSFER;
+
+      const id = moment().format('x');
+      dispatchTransaction(TransactionActionCreators.setTransactionRunning(true));
+
+      try {
+        const transfer = sourceApi.tx.balances.transfer(receiverAddress, transferAmount);
+        const options: Partial<SignerOptions> = {
+          nonce: -1
+        };
+        let sourceAccount: string | KeyringPair = account;
+        if (account.meta.isInjected) {
+          const injector = await web3FromSource(account.meta.source as string);
+          options.signer = injector.signer;
+          sourceAccount = account.address;
+        }
+
+        const unsub = await transfer.signAndSend(sourceAccount, { ...options }, ({ status }) => {
+          if (status.isReady) {
+            dispatchTransaction(
+              TransactionActionCreators.createTransactionStatus({
+                block: null,
+                blockHash: null,
+                id,
+                input: transferAmount,
+                messageNonce: null,
+                receiverAddress,
+                sourceAccount: account.address,
+                sourceChain,
+                status: TransactionStatusEnum.IN_PROGRESS,
+                targetChain: '',
+                type,
+                payloadHex: '',
+                transactionDisplayPayload: {} as TransactionDisplayPayload
+              })
+            );
+          }
+
+          if (status.isBroadcast) {
+            dispatchMessage(MessageActionsCreators.triggerInfoMessage({ message: 'Transaction was broadcasted' }));
+            dispatchTransaction(TransactionActionCreators.reset());
+          }
+
+          if (status.isInBlock) {
+            sourceApi.rpc.chain
+              .getBlock(status.asInBlock)
+              .then((res) => {
+                const block = res.block.header.number.toString();
+                dispatchTransaction(
+                  TransactionActionCreators.updateTransactionStatus(
+                    {
+                      block,
+                      blockHash: status.asInBlock.toString(),
+                      status: TransactionStatusEnum.COMPLETED
+                    },
+                    id
+                  )
+                );
+              })
+              .catch((e) => {
+                logger.error(e.message);
+                throw new Error('Issue reading block information.');
+              });
```

Comment:
Since this is included in an async function - I would alter this to async/await

```suggestion
            try {
              const res = await sourceApi.rpc.chain.getBlock(status.asInBlock)
              const block = res.block.header.number.toString();
              dispatchTransaction(
                TransactionActionCreators.updateTransactionStatus(
                  {
                      block,
                      blockHash: status.asInBlock.toString(),
                      status: TransactionStatusEnum.COMPLETED
                    },
                    id
                 )
                );
              } catch (e) {
                logger.error(e.message);
                throw new Error('Issue reading block information.');
              };
```

Reply:
YES !

Reply:
YES!